### PR TITLE
Update xgb algo files

### PR DESCRIPTION
**Pull request by qbc2016**
- refine each client's test data
 - refine testing procedure
 - add 3 more datasets for xgb algo

### CODE REVIEW

File: federatedscope/vertical_fl/xgb_base/worker/XGBClient.py [hunk 1]
```diff
@@ -48,6 +48,13 @@ def __init__(self,
 
         self.data = data
         self.own_label = ('y' in self.data['train'])
+
+        self.test_x = self.data['test']['x']
+        if self.own_label:
+            self.test_y = self.data['test']['y']
+
+        self.test_z = np.zeros(self.test_x.shape[0])
```
Comment (xieyxclack) [inline federatedscope/vertical_fl/xgb_base/worker/XGBClient.py:56]: what `test_z` means?

**Review by xieyxclack — COMMENTED** [review]
LGTM, please refer to inline comments for more suggestions

File: federatedscope/vertical_fl/dataset/abalone.py [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+    Height / continuous / mm / with meat in shell
+    Whole weight / continuous / grams / whole abalone
+    Shucked weight / continuous / grams / weight of meat
+    Viscera weight / continuous / grams / gut weight (after bleeding)
+    Shell weight / continuous / grams / after being dried
+    Rings / integer / -- / +1.5 gives the age in years
+
+    Arguments:
+        root (str): root path
+        name (str): name of dataset, ‘adult’ or ‘xxx’
```
[the remark this answers is not in the document]
Comment (qbc2016) [reply federatedscope/vertical_fl/dataset/abalone.py:39]: Ah, sry.., it should be 'blog'

File: federatedscope/vertical_fl/xgb_base/worker/XGBClient.py (hunk 1 above)
Comment (qbc2016) [reply federatedscope/vertical_fl/xgb_base/worker/XGBClient.py:56]: It is exactly the result outputted by the model on test data, I'll change the name.

**Review by xieyxclack — APPROVED** [review]

**Pull request merged**